swap/src/asb/event_loop.rs

```diff
@@ -319,13 +319,33 @@ where
         min_buy: bitcoin::Amount,
         max_buy: bitcoin::Amount,
     ) -> Result<BidQuote> {
-        let rate = self
+        let ask_price = self
             .latest_rate
             .latest_rate()
-            .context("Failed to get latest rate")?;
+            .context("Failed to get latest rate")?
+            .ask()
+            .context("Failed to compute asking price")?;
+
+        let max_bitcoin_for_monero = self
+            .monero_wallet
+            .get_balance()
+            .await?
+            .max_bitcoin_for_price(ask_price);
+
+        if max_buy > max_bitcoin_for_monero {
+            tracing::warn!(
+                    "Your Max bitcoin to sell config: {} is bigger than your current balance of Monero available to sell {}",
+                    max_buy, max_bitcoin_for_monero
+                );
+            return Ok(BidQuote {
+                price: ask_price,
+                min_quantity: min_buy,
+                max_quantity: max_bitcoin_for_monero,
+            });
+        }
 
```

Comment:
If max_bitcoin_buy is less than min_buy, the user cannot initiate a swap at all. We need to handle this case somehow. I suggest that we set both min_quantity and max_quantity to zero to communicate to the user that the ASB is not accepting swaps at the moment. Otherwise there might be a weird case where for example `min_quantity = 1 BTC` and `max_quantity = 0.5 BTC` which doesn't really make sense.

Something like this:
```rust
        if min_buy > max_bitcoin_for_monero {
             tracing::warn!(
                     "Your Monero balance is too low to initiate a swap, as your minimum swap amount is {}. You could at most swap {}",
                     min_buy, max_bitcoin_for_monero
                 );
             return Ok(BidQuote {
                 price: ask_price,
                 min_quantity: bitcoin::Amount::ZERO,
                 max_quantity: bitcoin::Amount::ZERO,
             });
         }
```

Reply:
I will wait for @thomaseizinger  response on the solution before upload, but it seems fine to me

Reply:
Yep, seems reasonable to me too. How does the swap tool handle both amounts being 0? Did someone do a test run for this?

Reply:
I didn't attempt a trade but this where the outputs that  it gave me. 
![Screenshot from 2022-03-01 09-05-42-cropped](https://user-images.githubusercontent.com/47948594/156130492-0c5c0898-f76f-47d6-845f-2ba98fe604f1.jpg)

Reply:
Can you attempt a trade? :)

Reply:
Okeyy, this are the results: The first time I tried to make a swap I't actually gave me a qr code to deposit and I tried it, as soon as the bitcoin arrived it failed and I received this message: 
![Screenshot from 2022-03-01 11-05-47-cropped](https://user-images.githubusercontent.com/47948594/156150064-1b8e2443-426d-43dd-84fa-3d3b58127203.jpg)

Then if I tried to do a swap again it wouldn't even let me swap it would just send the error again.
![Screenshot from 2022-03-01 11-07-23](https://user-images.githubusercontent.com/47948594/156150190-29411670-fe6a-4463-a055-6f8f1b384c58.png)

The Bitcoin from the first transaction redeemed correctly

Reply:
Okay, so the behaviour here is that we try to do a swap with the bitcoin already stored in the wallet and don't have you deposit new coins.

I think together with this change, we should add a check to the client whether or not `maximum_amount` is `0` and if so, abort directly.

Reply:
Done!